tests/conftest.py

```diff
@@ -42,3 +45,21 @@ def stub_v1(grpc_channel: grpc.Channel) -> v1_measurement_service_pb2_grpc.Measu
 def stub_v2(grpc_channel: grpc.Channel) -> v2_measurement_service_pb2_grpc.MeasurementServiceStub:
     """Test fixture that creates a MeasurementService v2 stub."""
     return v2_measurement_service_pb2_grpc.MeasurementServiceStub(grpc_channel)
+
+
+@pytest.fixture(scope="session")
+def discovery_service_process() -> Generator[DiscoveryServiceProcess, None, None]:
+    """Test fixture that creates discovery service process."""
+    if sys.platform != "win32":
+        pytest.skip(
+            f"Platform {sys.platform} is not supported for starting discovery service."
+            "Could not proceed to run the tests."
+        )
+    elif not pathlib.Path.exists(_get_registration_json_file_path()):
```

Comment:
```suggestion
    elif not pathlib.Path(_get_registration_json_file_path()).exists():
```

This is if _get_registration_json_file_path() returns a str, if it returns a Path object, just do

```suggestion
    elif not _get_registration_json_file_path().exists():
```

Reply:
The return type is a `Path` object, So I have used the second suggested change.